app/src/behaviors/behavior_sticky_key.c

```diff
@@ -184,44 +186,41 @@ static int sticky_key_keycode_state_changed_listener(const zmk_event_t *eh) {
     }
     for (int i = 0; i < ZMK_BHV_STICKY_KEY_MAX_HELD; i++) {
         struct active_sticky_key *sticky_key = &active_sticky_keys[i];
-        if (sticky_key->position == ZMK_BHV_STICKY_KEY_POSITION_FREE) {
+        if (sticky_key->trace_id == ZMK_BHV_STICKY_KEY_UNUSED) {
             continue;
         }
 
-        if (strcmp(sticky_key->config->behavior.behavior_dev, "KEY_PRESS") == 0 &&
-            HID_USAGE_ID(sticky_key->param1) == ev->keycode &&
-            (HID_USAGE_PAGE(sticky_key->param1) & 0xFF) == ev->usage_page &&
-            SELECT_MODS(sticky_key->param1) == ev->implicit_modifiers) {
-            // don't catch key down events generated by the sticky key behavior itself
+        if (sticky_key->trace_id == ev->trace_id) {
+            // don't catch events generated by the sticky key behavior itself
             continue;
         }
 
         // If events were queued, the timer event may be queued late or not at all.
         // Release the sticky key if the timer should've run out in the meantime.
         if (sticky_key->release_at != 0 && ev->timestamp > sticky_key->release_at) {
             stop_timer(sticky_key);
+            LOG_DBG("releasing 1");
```

Comment:
Should these messages be left in? I'm not sure what the significance of release 1, 2, and 3 are.

Reply:
whoops